supabase/migrations/03_catalog-types.sql

```diff
@@ -24,6 +24,20 @@ For example: "acmeCo/anvils/" or "acmeCo/products/TnT_v4/",
 but not "acmeCo/anvils" or "acmeCo/some anvils".
 ';
 
+create domain catalog_tenant as text
+  constraint "Must be a valid catalog tenant"
+  check (value ~ '^[[:alpha:][:digit:]\-_.]+/$' and value is nfkc normalized);
```

Comment:
Doesn't have to be right now, but I think we should consider additional restrictions on tenant names. As it is, I don't see how we'd disallow names like `../` or `-/`, which seem likely to cause issues. I'm also still skeptical of allowing both `foo/` and `Foo/`, since it could easily lead to confusion for users.

Reply:
The `tenant_exists` check does case-invariant comparison, so you can't have `foo/` and `Foo/`. Once provisioned, the tenant is case sensitive though.

We can also "reserve" tenants by creating them in the `tenants` table without being hooked up to anything.